apis/v1alpha2/nginxproxy_types.go

```diff
@@ -72,12 +72,35 @@ type NginxProxySpec struct {
 	//
 	// +optional
 	DisableHTTP2 *bool `json:"disableHTTP2,omitempty"`
+	// WAF enables NGINX App Protect WAF functionality.
+	// When enabled, NGINX Gateway Fabric will deploy additional WAF containers
+	// (waf-enforcer and waf-config-mgr) alongside the main NGINX container.
+	// Default is "Disabled".
+	//
+	// +optional
+	// +kubebuilder:default:=Disabled
+	WAF *WAFState `json:"waf,omitempty"`
 	// Kubernetes contains the configuration for the NGINX Deployment and Service Kubernetes objects.
 	//
 	// +optional
 	Kubernetes *KubernetesSpec `json:"kubernetes,omitempty"`
 }
 
+// WAFState defines the state of WAF functionality.
+//
+// +kubebuilder:validation:Enum=Enabled;Disabled
```

Comment:
for consistency I think we want these enums to be lowercase.

Reply:
ugh I don't remember why we do that...the image pull policies are based off of the k8s values, but the rest I'm not sure.